gems/pending/appliance_console/prompts.rb

```diff
@@ -85,6 +85,21 @@ def ask_for_ip_or_hostname_or_none(prompt, default = nil)
       ask_for_ip(prompt, default, validation, "a valid Hostname or IP Address.").gsub(/^'?NONE'?$/i, "")
     end
 
+    def ask_for_schedule_frequency(prompt = "Select schedule frequency: hourly, daily, weekly or monthly", default = "hourly")
+      validation = ->(h) { h.empty? || h =~ /^hourly$/i || h =~ /^daily$/i || h =~ /^weekly$/i || h =~ /^monthly$/i }
+      just_ask(prompt, default, validation, "hourly, daily, weekly or monthly")
+    end
+
+    def ask_for_hour_number(prompt = "Select hour(0-23)", default = "0")
+      validation = ->(h) { h.empty? || h =~ /^(#{(0..23).to_a * '|'})$/ }
+      just_ask(prompt, default, validation, "hour(0-23)")
+    end
+
+    def ask_for_day_number(prompt = "Select day(0-6)", default = "0")
+      validation = ->(h) { h.empty? || h =~ /^(#{(0..6).to_a * '|'})$/ }
```

Comment:
@carbonin Yes I agree. We do want empty input to fail so the user is prompted.

Also in testing  I noticed we need to test if the input is an integer. So the test becomes:

```ruby
validation = ->(h) { h =~ INT_REGEXP && (0..6).include?(h.to_i) }
```

This eliminates the need to explicitly test for ` !  h.empty? ` because if h is empty is is not an integer.